### ISSUE TITLE
Add better mouse support

### ISSUE THREAD

**Issue by niutech** [mason-larobina/luakit#283]
I would be great to make Luakit more (regular) user friendly by improving mouse support, that is:

 - clicking on the address bar to type a new URL
 - adding [X] on each tab which closes a tab on click
 - adding [+] button which opens a new tab on click
 - right clicking on selected text to search using the default search provider
 - right clicking on link to open it in the new background tab

Are there any plugins which cover all those features? Thanks!

**Comment by o-jasper** [comment]
I think this would need changing `config/window.lua`.

What i would prefer personally is to go "html-only" on any non-basic gui. Basically, make multiple pages side-by-side work, and then make chrome-pages that fill them. That way, it should be very easy and consistent-with-chrome-pages to make/change them. At the same time we might make something like add-bookmark just show on the side, instead of filling a new tab.

**Comment by martingabelmann** [comment]
Im not sure if the first 3 features are the Luakit-way...
  * Open uris in backgroundtab works either with 'F + hint' or  'Ctrl + Mouseclick'.
  * in general selected text is within the primary clipboard. you can either press 'P' oder the middle mouse to open it (if its not a valid uri Luakit is opening yourch searchengine with the copied string).

**Comment by o-jasper** [comment]
@martingabelmann agree. But then i am hoping it to be very customizable, and that can be used to create a user -as in non-techy, non-configuring- -friendly version.

**Comment by aidanholm** [comment]
> clicking on the address bar to type a new URL

Best done by modifying (a copy of) `lib/lousy/widget/uri.lua`.

> adding [X] on each tab which closes a tab on click, adding [+] button which opens a new tab on click

Doable with a `build` signal handler; see `lib/tab_favicons.lua` for an example of that.

> right clicking...

Should be doable with binds in `config/binds.lua`

**Comment by groolot** [comment]
I'm not sure that a keyboard driven web browser has to be more mouse friendly. I think we should concentrate efforts on enhancing performance and stability, not mouse interactiion.

**Comment by aidanholm** [comment]
Sure, I agree; I don't have any intention of adding these features myself. But all these ideas can be done with 5-10 lines of code, and it could go in the wiki.

**Comment by arteze** [comment]
In `window.lua` in `tablist_tab_click = function (w)` change:

```lua
elseif button == 2 or button == 3 then
```

The 2 is middle click, the 3 is right click.
